samples/zigbee/light_switch/src/main.c

```diff
@@ -169,25 +173,18 @@ static void button_handler(u32_t button_state, u32_t has_changed)
 	case BUTTON_ON:
 	case BUTTON_OFF:
 		LOG_DBG("Button pressed");
-		atomic_set(&device_ctx.button.in_progress, ZB_TRUE);
-		zb_err_code = zigbee_schedule_alarm(light_switch_button_handler,
-						    button_state,
-						    BUTTON_LONG_POLL_TMO);
-		if (zb_err_code == RET_OVERFLOW) {
-			LOG_WRN("Can't schedule another alarm, queue is full.");
-			atomic_set(&device_ctx.button.in_progress, ZB_FALSE);
-		} else {
-			ZB_ERROR_CHECK(zb_err_code);
-		}
+		atomic_set(&buttons_ctx.in_progress, ZB_TRUE);
+		buttons_ctx.state = button_state;
+		k_timer_start(&buttons_ctx.alarm, BUTTON_LONG_POLL_TMO,
+			      K_NO_WAIT);
 		break;
 	case 0:
 		LOG_DBG("Button released");
 
-		ZB_SCHEDULE_APP_ALARM_CANCEL(light_switch_button_handler,
-					     ZB_ALARM_ANY_PARAM);
-		atomic_set(&device_ctx.button.in_progress, ZB_FALSE);
+		k_timer_stop(&buttons_ctx.alarm);
+		atomic_set(&buttons_ctx.in_progress, ZB_FALSE);
 
-		if (atomic_set(&device_ctx.button.long_poll, ZB_FALSE)
+		if (atomic_set(&buttons_ctx.long_poll, ZB_FALSE)
```

Comment:
In case of a following scenario:
 - Press BUTTON_ON
 - Press BUTTON_OFF
 - Release BUTTON_ON
 - Press BUTTON_ON
 - Release BUTTON_ON
 - Press BUTTON_ON
 - Release BUTTON_OFF

Strange things will happen. This case might be the reason, why the initial implementation was using the `in_progress` flag.

Reply:
True, but I've wrote this code looking at behavior of Ikea switch, where is not possible to press ON OFF together. If we want to handle more general cases I will change this code.

Reply:
Well - I do not say that it should do anything when more buttons are pressed. It may simply ignore those pressed. From the code it looked like pressing more buttons will create more Zephyr alarms and send more packets to the bulb, which is a different behavior than the one you can find in IKEA switches.

Reply:
Only one Zephyr alarm can be scheduled. So pressing more buttons at the same time shouldn't be a problem.